hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/impl/TokenGetCustomFeesPrecompile.java

```diff
@@ -66,8 +66,8 @@ public Builder body(final Bytes input, final UnaryOperator<byte[]> aliasResolver
 
     @Override
     public Bytes getSuccessResultFor(final ExpirableTxnRecord.Builder childRecord) {
-        validateTrue(stateView.tokenExists(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);
-        final var customFees = stateView.tokenCustomFees(tokenId);
+        validateTrue(ledgers.tokens().contains(tokenId), ResponseCodeEnum.INVALID_TOKEN_ID);
+        final var customFees = ledgers.tokenCustomFees(tokenId);
```

Comment:
If we make the method to return an Optional we can validate the token by validating the Optional result and skip the tokens().contains() check.